text/0000-global-allocators.md

```diff
@@ -240,9 +252,47 @@ The allocator APIs could be simplified to a more "traditional"
 malloc/calloc/free API at the cost of an efficiency loss when using allocators
 with more powerful APIs.
 
+The global allocator could be an instance of the `Allocator` trait. Since that
+trait's methods take `&mut self`, things are a bit complicated however. The
+allocator would most likely need to be a `const` type implementing `Allocator`
+since it wouldn't be sound to interact with a static. This may cause confusion
```

Comment:
I agree completely. Allocators are inherently stateful since they need to keep track of allocations for correctness. Static + interior mutability is needed.

However, this raises a new question: initializing the global allocator. How does this happen? Is there a special constructor called? Does the constructor have to be a const fn? The RFC doesn't specify this.